src/txn/txn_ckpt.c

```diff
@@ -640,9 +640,8 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_ASSERT(session, session->ckpt_handle_next == 0);
 	WT_WITH_SCHEMA_LOCK(session,
 	    WT_WITH_TABLE_LOCK(session,
-		WT_WITH_HANDLE_LIST_LOCK(session,
-		    ret = __checkpoint_apply_all(
-		    session, cfg, __wt_checkpoint_get_handles, NULL))));
+		ret = __checkpoint_apply_all(
```

Comment:
Will switching to a read-lock that is released during the traverse mean that checkpoint can potentially retrieve an inconsistent set of handles during this traverse? The traverse is done before the checkpoint transaction has started, so maybe this change doesn't make anything less consistent than it was before? Or does the schema lock protect against parallel operations that would interfere?